android/src/main/java/com/swmansion/gesturehandler/RNGestureHandlerPackage.kt

```diff
@@ -1,21 +1,84 @@
 package com.swmansion.gesturehandler
 
-import com.facebook.react.ReactPackage
+import com.facebook.react.TurboReactPackage
+import com.facebook.react.ViewManagerOnDemandReactPackage
+import com.facebook.react.bridge.ModuleSpec
 import com.facebook.react.bridge.NativeModule
 import com.facebook.react.bridge.ReactApplicationContext
+import com.facebook.react.module.annotations.ReactModule
+import com.facebook.react.module.model.ReactModuleInfo
+import com.facebook.react.module.model.ReactModuleInfoProvider
+import com.facebook.react.turbomodule.core.interfaces.TurboModule
+import com.facebook.react.uimanager.ReactShadowNode
 import com.facebook.react.uimanager.ViewManager
 import com.swmansion.gesturehandler.react.RNGestureHandlerButtonViewManager
 import com.swmansion.gesturehandler.react.RNGestureHandlerModule
 import com.swmansion.gesturehandler.react.RNGestureHandlerRootViewManager
 
-class RNGestureHandlerPackage : ReactPackage {
-  override fun createNativeModules(reactContext: ReactApplicationContext): List<NativeModule> {
-    return listOf<NativeModule>(RNGestureHandlerModule(reactContext))
+class RNGestureHandlerPackage : TurboReactPackage(), ViewManagerOnDemandReactPackage {
+  private val viewManagers: Map<String, ModuleSpec> by lazy {
+    mapOf(
+      RNGestureHandlerRootViewManager.REACT_CLASS to ModuleSpec.viewManagerSpec {
+        RNGestureHandlerRootViewManager()
+      },
+      RNGestureHandlerButtonViewManager.REACT_CLASS to ModuleSpec.viewManagerSpec {
+        RNGestureHandlerButtonViewManager()
+      }
+    )
   }
 
   override fun createViewManagers(reactContext: ReactApplicationContext) =
     listOf<ViewManager<*, *>>(
       RNGestureHandlerRootViewManager(),
       RNGestureHandlerButtonViewManager()
     )
+
+  override fun getViewManagerNames(reactContext: ReactApplicationContext?) =
+    viewManagers.keys.toList()
+
+  override fun getViewManagers(reactContext: ReactApplicationContext?): MutableList<ModuleSpec> =
+    viewManagers.values.toMutableList()
+
+  override fun createViewManager(
+    reactContext: ReactApplicationContext?,
+    viewManagerName: String?
+  ): ViewManager<*, out ReactShadowNode<*>>? {
```

Comment:
Why those shenanigans with `ReactShadowNode` are needed?

Reply:
They are not